documentation/ingestion_server/guides/migrate.md

```diff
@@ -0,0 +1,253 @@
+and will be made available to the API.
+
+### API-involved
+
+These changes are modifications to fields that already are in use by the API and
+involve code changes in both the ingestion server and the API. Examples:
+
+- removal of a field
+- changing the type to an incompatible type
+- renaming of a field
+
+Such kinds of changes need us to precisely deploy the API simultaneously with
+the promotion of new index because of these reasons:
+
+- If the API deployment lags behind index promotion, the old field that the API
+  uses will disappear.
+- If the API deployment leads ahead of index promotion, the new field the API
+  uses will not be present.
+
+This runbook documents guidelines and processes for API-involved migrations.
+
+Our goal is to break down an API-involved change into multiple small, atomic
+changes with each step affecting at most one of the ingestion server or the API
+and ensuring that the API and ES remain compatible throughout the process.
+
+## Pull request guidelines
+
+A change that involves modification to the ES index as well as its usage in the
+API requires at least three steps, each associated with exactly one PR that
+modifies exactly one of the ingestion server or the API to allow them to be
+deployed independently.
+
+1. Change the ES index mapping in the ingestion server. Ensure that the change
+   is purely additive, keeping the old fields unchanged and creating new fields
+   that contain the data the API will need.
+
+   This PR should make changes only within the `ingestion_server/` directory,
+   more specifically the following two files concerned with ES mappings and
+   document schemas:
+
+   - [`es_mappings.py`](https://github.com/WordPress/openverse/tree/main/ingestion_server/ingestion_server/es_mapping.py)
+   - [`elasticsearch_models.py`](https://github.com/WordPress/openverse/tree/main/ingestion_server/ingestion_server/elasticsearch_models.py)
+
+2. Update the API code to reference and use the new ES fields added in the
+   previous step. Ensure that the old fields become unreferenced.
+
+   The PR should make changes only within the `api/` directory.
+
+3. Change the ES index mapping in the ingestion server to remove the old,
+   now-unreferenced fields.
+
+   Like PR number 1, this PR should also make changes only within the
+   `ingestion_server/` directory.
+
+```{tip}
+Get the PRs reviewed in advance so that the entire process has been vetted by
+the team and there are no surprises or delays when the plans have been set into
+motion.
+```
+
+```{caution}
+Each PR in the chain should branch from, and point to, its predecessor in the
+chain so that CI continues to pass for each PR.
+```
+
+### Example
+
+Assume we have a field `foo` with type `text` in the index. It has a subfield
+`keyword` with type `keyword`. The API uses `foo.keyword` for all purposes.
+
+One PR that does these 3 things would be an API-involved change. So we split
+them into 3 PRs.
```

Comment:
What are the 3 things referenced in this line? Could you potentially reword this for clarity?

Reply:
Implemented in 66134fa.